core/unit_test/cuda/TestCuda_InterOp_StreamsMultiGPU.cpp

```diff
@@ -105,4 +105,17 @@ TEST(cuda_multi_gpu, scratch_space) {
     test_scratch(execs[0], execs[1]);
   }
 }
+
+TEST(cuda_multi_gpu, stream_sync_semantics) {
```

Comment:
Bigger picture question for this test, and I'd like to invite @dalg24 to share his though on this.

In this test, the initializer of `StreamsAndDevices` would use 2 streams on the same device (ID 0) if the executable only sees one device.

So these tests are also marked as passed on a single device machine.

I think they should be marked as **skipped*.

Reply:
This is fine with me.